tests/commands/test_api_error.py

```diff
@@ -0,0 +1,108 @@
+
+        result = self.cli("record", "build", "--name", "example")
+        self.assertEqual(result.exit_code, 0)
+
+    @responses.activate
+    @mock.patch.dict(os.environ, {"LAUNCHABLE_TOKEN": CliTestCase.launchable_token})
+    def test_record_session(self):
+        build = "internal_server_error"
+        responses.add(responses.POST, "{base}/intake/organizations/{org}/workspaces/{ws}/builds/{build}/test_sessions".format(
+            base=get_base_url(), org=self.organization, ws=self.workspace, build=build), status=500)
+
+        result = self.cli("record", "session", "--build", build)
+        self.assertEqual(result.exit_code, 0)
+
+        build = "not_found"
+        responses.add(responses.POST, "{base}/intake/organizations/{org}/workspaces/{ws}/builds/{build}/test_sessions".format(
+            base=get_base_url(), org=self.organization, ws=self.workspace, build=build), status=404)
+
+        result = self.cli("record", "session", "--build", build)
+        self.assertEqual(result.exit_code, 1)
+
+    @responses.activate
+    @mock.patch.dict(os.environ, {"LAUNCHABLE_TOKEN": CliTestCase.launchable_token})
+    def test_subset(self):
+        responses.replace(responses.POST, "{base}/intake/organizations/{org}/workspaces/{ws}/subset".format(
+            base=get_base_url(), org=self.organization, ws=self.workspace), status=500)
+
+        subset_file = "example_test.rb"
+
+        with tempfile.NamedTemporaryFile(delete=False) as rest_file_500:
+            result = self.cli("subset", "--target", "30%", "--session", self.session, "--rest", rest_file_500.name,
+                              "minitest", str(self.test_files_dir) + "/test/**/*.rb", mix_stderr=False)
+
+            self.assertEqual(result.exit_code, 0)
+            self.assertEqual(len(result.stdout.rstrip().split("\n")), 1)
+            self.assertTrue(subset_file in result.stdout)
+
+            rest = Path(rest_file_500.name).read_text()
+            self.assertEqual(
+                len(rest.rstrip().split("\n")), 1)
+            self.assertTrue(subset_file in rest)
+
+        responses.replace(responses.POST, "{base}/intake/organizations/{org}/workspaces/{ws}/subset".format(
+            base=get_base_url(), org=self.organization, ws=self.workspace), status=404)
+
+        rest_file_404 = tempfile.NamedTemporaryFile(delete=False)
+        result = self.cli("subset", "--target", "30%", "--session", self.session, "--rest", rest_file_404.name,
+                          "minitest", str(self.test_files_dir) + "/test/**/*.rb", mix_stderr=False)
+        self.assertEqual(result.exit_code, 0)
+
+        self.assertEqual(len(result.stdout.rstrip().split("\n")), 1)
+        self.assertTrue(subset_file in result.stdout)
+
+        rest = Path(rest_file_404.name).read_text()
+        self.assertEqual(
+            len(rest.rstrip().split("\n")), 1)
+        self.assertTrue(subset_file in rest)
+
+        rest_file_404.close()
+        os.unlink(rest_file_404.name)
```

Comment:
[nits]
this can be written with `with` clause to avoid forgetting to delete named tmp file.
```python
        with tempfile.NamedTemporaryFile() as rest_file_404:
                result = self.cli("subset", "--target", "30%", "--session", self.session, "--rest", rest_file_404.name,
                                  "minitest", str(self.test_files_dir) + "/test/**/*.rb", mix_stderr=False)
                self.assertEqual(result.exit_code, 0)

                self.assertEqual(len(result.stdout.rstrip().split("\n")), 1)
                self.assertTrue(subset_file in result.stdout)

                rest = Path(rest_file_404.name).read_text()
                self.assertEqual(len(rest.rstrip().split("\n")), 1)
                self.assertTrue(subset_file in rest)
```